### ISSUE TITLE
Add Zstd Custom Compression Level Option

### ISSUE THREAD

**Issue by JoshStrobl** [ajour/ajour#672]
The backup compression / option for Zstd uses the default compression level (default: 3). With my specific backup folder, which is 83.9MiB, the zstd compression will create a 17.1MiB file. This is pretty fantastic given it performs this compression in 300ms, however I would like to request that the ability to set our own compression level. In terms of the UX, I imagine this option could be exposed in the most straightforward manner with some form of a compression slider. I threw a mockup together in Inkscape to illustrate it.

![ajour-compression-slider](https://user-images.githubusercontent.com/156574/124328595-99625a80-db92-11eb-9a18-7be5c8ec53f5.png)

Here are some metrics I ran for some higher zstd compression levels. This is an average of 5 runs.

Compression Level | Time | Resulting File Size
---- | ---- | ----
-8 | 1.158s | 15.8MiB
-13 | 4.946s | 15.4MiB
-18 | 17.872s | 14.4MiB
-19 | 27.04s | 14.1MiB
-22 (with --ultra) | 45.878s | 13.8MiB

The zstd rust package provides a mechanism to update the `CompressionLevel` for the encoder via [set_parameter](https://docs.rs/zstd/0.9.0+zstd.1.5.0/zstd/stream/write/struct.Encoder.html#method.set_parameter).

Assuming the implementation of the mockup above, I would propose:

- Fastest: 3 (default)
- Faster: 8
- Fast: 13
- Slow: 18
- Slowest: 22

**Comment by casperstorm** [comment]
Good idea, @JoshStrobl.
The suggested UI is a good starting point, a little too dominant if you ask me but the idea is great.
I would suggest also only showing the if `zstd` is selected.

**Comment by JoshStrobl** [comment]
Yea I agree that it should only be displayed if zstd is selected. In terms of the slider, there is certainly room for improvement, whether that is reducing the overall thickness of the slider to reducing its width as a couple examples. I am not sure what sort of graphical toolkit you are using, so not really sure what is at your disposal.

**Comment by tarkah** [comment]
Some helpful documentation since we've not implemented a slider yet...

- [iced_native::widget::slider::Slider](https://docs.rs/iced_native/0.4.0/iced_native/widget/slider/struct.Slider.html)
  - We can set the `step` size to 1 and `range` as the number of options
  - I'm assuming the slider will "snap" to each step location (even increments of the width)
- [iced_style::slider::Style](https://docs.rs/iced_style/0.3.0/iced_style/slider/struct.Style.html)
  - We can set the active / inactive rail colors as well as style the handle.
- I'm assuming a row below the slider with text elements = # of options will align well to the step / snap points

**Comment by Bananaman** [comment]
Nice idea to have better compression... Why not just use a drop down menu with the choices? :D

**Comment by JoshStrobl** [comment]
Because solely having a dropdown does a poor job communicating why something may be labeled "Slowest".

**Comment by casperstorm** [comment]
I've made some good progress on this feature. I'll finish it later this week.

<img width="717" alt="Screenshot 2021-07-12 at 00 06 27" src="https://user-images.githubusercontent.com/2248455/125211188-0669a480-e2a5-11eb-84b2-745131333898.png">

**Comment by JoshStrobl** [comment]
@casperstorm That looks really great! Looking forward to trying it out,

**Comment by casperstorm** [comment]
Released in [1.2.3-rc.1](https://github.com/ajour/ajour/releases/tag/1.2.3-rc.1).

**Comment by Bananaman** [comment]
@casperstorm @JoshStrobl The feature works great!

![image](https://user-images.githubusercontent.com/38923130/126072028-9f9e5c7e-8cc7-454d-897d-6d3a7fa409be.png)

I found that this exact spot on the slider was optimal on a Ryzen 3900x. It results in ~12 seconds to compress, generating an archive that's: **80,5 MB (84 459 464 bytes)**

From these files: **371 MB (389 887 548 bytes), 8 356 Files, 1 307 Folders**

That's an impressive speed and compression ratio.

At max compression on the slider, it instead took 150 seconds which was way too much, and generated a file that's 70,2 MB (73 704 828 bytes). So only a tiny bit more savings than the "12 seconds" version I settled on, but an unacceptable amount of time spent.

I hope this is good advice for others. Just set your slider somewhere around the "Slower" mark. Not the "Slowest" choice. We're talking a few megabytes difference at a cost of many minutes of tedious waiting. ;-)

Anyway, this feature is great, thanks everyone!
